File: typescript/nucleotide-count/nucleotide-count.ts
export default class NucleotideCount {

  static nucleotideCounts(nucleotides: string) {

    let count = {
      'A': 0,
      'C': 0,
      'G': 0,
      'T': 0
    }

    for (let i = 0; i < nucleotides.length; i++) {
      let invalid: boolean = false
      nucleotides[i] === 'A' ? count.A += 1 :
        nucleotides[i] === 'C' ? count.C += 1 :
          nucleotides[i] === 'G' ? count.G += 1 :
            nucleotides[i] === 'T' ? count.T += 1 :
              invalid = true
      if (invalid) { throw new Error('Invalid nucleotide in strand.') }
    }
    return count
  }
}

File: csharp/nucleotide-count/NucleotideCount.cs
using System;
using System.Collections.Generic;

public static class NucleotideCount
{
    public static IDictionary<char, int> Count(string sequence)
    {
        Dictionary<char, int> nucleotideCount = new Dictionary<char, int>
        {
            { 'A', 0 },
            { 'C', 0 },
            { 'G', 0 },
            { 'T', 0 }
        };

        foreach (char nucleotide in sequence)
        {
            if (nucleotideCount.ContainsKey(nucleotide))
            {
                nucleotideCount[nucleotide]++;
            }
            else
            {
                throw new System.ArgumentException();
            }
        }
        return nucleotideCount;
    }
}
